tests/lib/string_conv/testcase.yaml

```diff
@@ -0,0 +1,6 @@
+tests:
+  libraries.string_conv:
+    tags: string to numeric conversion
```

Comment:
Some qemu's has no FPU. I would recommend following: 
1. Turn this test case to one with FPU (remove `CONFIG_FPU=y` from `prj.conf`).
```suggestion
  tags: string to numeric conversion
  filter: CPU_HAS_FPU
  extra_configs:
    - CONFIG_FPU=y
```

2. Add second test case without fpu. This should not need any filter.

In `main.c` add ifdef to fpu cases.

Reply:
It was not only the FPU option that was the issue. I tried to filter the test to only run on devices that contained FPU, but that did not resolve this. All three tests for str2long, str2ulong and str2dbl failed. on basis of the feedback from these tests I suspect that it could be an errno related issue for the qemu platforms, but im not certain.